### ISSUE TITLE
data_format="NCHW"    didn't   work  for    tf.nn.max_pool

### ISSUE THREAD

**Issue by guicunbin** [tensorflow/tensorflow#12999]
In [78]: x2 = tf.random_uniform([64,3,32,32])

In [79]: x2_pool = tf.nn.max_pool(x2,  [1,3,3,1], [1,2,2,1], padding="VALID", data_format="NHWC")

In [80]: x2_pool.shape
Out[80]: TensorShape([Dimension(64), Dimension(1), Dimension(15), Dimension(32)])

In [81]: x2_pool = tf.nn.max_pool(x2,  [1,3,3,1], [1,2,2,1], padding="VALID", data_format="NCHW")

In [82]: x2_pool.shape
Out[82]: TensorShape([Dimension(64), Dimension(1), Dimension(15), Dimension(32)])

**Comment by calledbymountains** [comment]
Isn't the above question more suited for SO ?

**Comment by calledbymountains** [comment]
You have to provide ksize and strides in the same format as the data. 

```
In [3]: x2 = tf.random_uniform([64,3,32,32])

In [4]: x2_pool = tf.nn.max_pool(x2, [1,3,3,1], [1,2,2,1], padding="VALID", data_format="NHWC")

In [5]: x2_pool.shape
Out[5]: TensorShape([Dimension(64), Dimension(1), Dimension(15), Dimension(32)])

In [6]: x2_pool = tf.nn.max_pool(x2, [1,3,3,1], [1,2,2,1], padding="VALID", data_format="NCHW")

In [7]: x2_pool.shape
Out[7]: TensorShape([Dimension(64), Dimension(1), Dimension(15), Dimension(32)])

In [8]: x2_pool = tf.nn.max_pool(x2, [1, 1, 3,3], [1, 1, 2,2], padding="VALID", data_format="NCHW")

In [9]: x2_pool.shape
Out[9]: TensorShape([Dimension(64), Dimension(3), Dimension(15), Dimension(15)])

```

**Comment by guicunbin** [comment]
thank you very much
